src/keys/slip10.rs

```diff
@@ -463,7 +475,7 @@ impl Chain {
     }
 
     pub fn all_non_hardened(&self) -> bool {
-        self.0.iter().all(|s| !s.is_hardened())
+        self.0.iter().all(Segment::is_non_hardened)
```

Comment:
More idiomatic to do
```
!self.0.iter().any(Segment::is_hardened)
```

Reply:
Thanks, already added `Segment::is_non_hardened`.

Reply:
Done.